src/libs/antares/benchmarking/DurationCollector.cpp

```diff
@@ -0,0 +1,24 @@
+#include <numeric>
+#include <string>
+#include "antares/benchmarking/DurationCollector.h"
+
+namespace Benchmarking {
+
+void DurationCollector::addDuration(const std::string& name, int64_t duration)
+{
+    const std::lock_guard<std::mutex> lock(mutex_);
+    duration_items_[name].push_back(duration);
+}
+
+void DurationCollector::toFileContent(FileContent& file_content)
+{
+    for (const auto& element : duration_items_)
+    {
+        const std::string& name = element.first;
+        const auto& durations = element.second;
```

Comment:
Not really the goal of this PR, but
```suggestion
    for (const auto& [name, durations] : duration_items_)
    {
```